main.cpp

```diff
@@ -67,10 +67,17 @@ int main(int argc, char ** argv)
     Configuration config;
     // TODO remove me in refactoring
     IniFileParser iniFileParser(&config);
-    if(!iniFileParser.Parse(a.IniFilePath())) {
+    try {
```

Comment:
Ugh... Now you have two ways to indicate that the inifile could not be parsed, a return value of false and an exception.

Either, Or but not both please :) Id say `Parse()` should catch everything and return true/false.

Reply:
I will add Exception error-handling in IniFileParser in another PR.